conjur/logic/login_logic.py

```diff
@@ -39,24 +42,19 @@ def get_api_key(cls, ssl_verify:bool, credential_data:CredentialsData,
             'account': conjurrc.conjur_account
         }
 
-        if ssl_verify is False:
-            certificate_path = False
-        elif ssl_verify and credential_data.machine.startswith("https"):
-            # Catches the case where a user does not run in insecure mode but the
-            # .conjurrc cert_file entry is empty
-            if conjurrc.cert_file == '':
-                raise CertificateVerificationException
-
-            certificate_path = conjurrc.cert_file
-
         # pylint: disable=logging-fstring-interpolation
         logging.debug(f"Attempting to fetch '{credential_data.login}' API key from Conjur...")
-        api_key = invoke_endpoint(HttpVerb.GET,
-                                  ConjurEndpoint.LOGIN,
-                                  params,
-                                  auth=(credential_data.login, password),
-                                  ssl_verify=certificate_path).text
-
+        try:
+            api_key = invoke_endpoint(HttpVerb.GET,
+                                      ConjurEndpoint.LOGIN,
+                                      params,
+                                      auth=(credential_data.login, password),
+                                      ssl_verification_metadata=ssl_verification_metadata).text
+        except HttpSslError:
+            if not conjurrc.cert_file and not ssl_verification_metadata.is_insecure_mode:
+                raise CertificateVerificationException
+        except Exception:
+            raise
```

Comment:
seems redundant. no need to catch exception just to raise it, unless it is specific, but it's not the case here.

Reply:
Done